bbc/simorgh
# WSTEAM1-455: Key Points UX styling + A11y

Resolves JIRA [WSTEAM1-455](https://jira.dev.bbc.co.uk/browse/WSTEAM1-455)

Overall changes
======
- Apply styling and A11y A/Cs to Key Points on Live Page.

![Screenshot 2023-10-10 at 09 52 36](https://github.com/bbc/simorgh/assets/98817636/5a78e2d1-8ed7-42ce-a51e-3981092f542f)

Code changes
======
- Render Key Points using legacy Blocks container. Use Bulleted List containers for multiple key points and Paragraph for single key point.
- Do not render Key Points section if the object is null in the live page response. (Expected behavior from simorgh-bff)
- Also do not render Key Points if there are no list items. (I do not know if it's possible for editorial to do this)
- Get section title from translations (Translation work to be completed by another ticket. Have labelled with a comment).
- Apply styles
- Create types
- Add/ update unit tests

Testing
======
View UX and A11y on the storybook permalinks
- [Live Page](https://wsteam1-455-key-points--5d28eb3fe163f6002046d6fa.chromatic.com/?path=/story/pages-live-page--example)
- [Single Key Point](https://wsteam1-455-key-points--5d28eb3fe163f6002046d6fa.chromatic.com/?path=/story/new-components-live-page-key-points--single-key-point)
- [Multiple Key Points](https://wsteam1-455-key-points--5d28eb3fe163f6002046d6fa.chromatic.com/?path=/story/new-components-live-page-key-points--multiple-key-points)

See the response on the live page 
- Checkout branch `WSTEAM1-455-key-points`
- cd into the `ws-nextjs-app` directory and run `yarn dev`
- Visit: http://localhost:7081/pidgin/live/c7p765ynk9qt (with key points)
- Visit: http://localhost:7081/pidgin/live/cx747l228kjt (no key points)

Helpful Links
======
[A11y A/Cs](https://paper.dropbox.com/doc/Live-page-Accessibility-Acceptance-Criteria--CAMMPe~y7Pbh31cFtPqHlIdhAg-KZQODmAf9tvodUlxsAVnA#:uid=759918899368087143410619&h2=Summary-(Key-points))
[UX Designs](https://www.figma.com/file/mxshZlXkCPPIh0oQJvRLpZ/Live-summary-box---handover?type=design&node-id=1%3A2&mode=design&t=5bExtkdFBel67cGC-1)
[Screen Reader UX](https://www.figma.com/file/mxshZlXkCPPIh0oQJvRLpZ/Live-summary-box---handover?type=design&node-id=502-67539&mode=design&t=gSZo3mYpGwIJaUrz-0)

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
